python/pip_install/pip_repository.bzl

```diff
@@ -125,20 +126,65 @@ def _get_toolchain_unix_cflags(rctx):
     if not is_standalone_interpreter(rctx, rctx.attr.python_interpreter_target):
         return []
 
+    include_paths = []
+
+    is_clang = False
+    if c_compiler:
+        err = rctx.execute([c_compiler, "-v"])
+        out = err.stderr.strip()
+        is_clang = "clang" in out.split("\n")[0]
+
+        if is_clang:
+            args = ["-E", "-Wp,-v", "-xc", "-"]
+        else:
+            args = ["-E", "-Wp,-v", "-xc++", "-"]
+        err = rctx.execute([c_compiler] + args)
+        out = err.stderr.strip()
+        parsed = [line.strip() for line in out.split("\n") if line.startswith(" ")]
```

Comment:
Testing this out on a Mac (ARM, macOS 13.2.1), I found that `clang -Wp,-v -xc -` returns this output:
```
clang -cc1 version 14.0.0 (clang-1400.0.29.202) default target arm64-apple-darwin22.3.0
ignoring nonexistent directory "/usr/local/include"
ignoring nonexistent directory "/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/local/include"
ignoring nonexistent directory "/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/Library/Frameworks"
#include "..." search starts here:
#include <...> search starts here:
 /Library/Developer/CommandLineTools/usr/lib/clang/14.0.0/include
 /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include
 /Library/Developer/CommandLineTools/usr/include
 /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks (framework directory)
End of search list.
```
The last entry includes some extra junk after the path which ends up being passed verbatim to `cc -i` and breaking sdist compilation with errors like:
```
clang: error: no such file or directory: '(framework'; did you mean '-framework'?
```
This change seems to fix it:
```suggestion
        parsed = [line.strip().partition(" ")[0] for line in out.split("\n") if line.startswith(" ")]
```

Reply:
@f0rmiga can we get this change in?